tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,516 @@
+    )
+    assert response.status_code == 200 and response.get_json()["data"] == expected
+
+
+def assert_registering_existing_queue_name_fails(
+    client: FlaskClient, name: str
+) -> None:
+    """Assert that registering a queue with an existing name fails.
+
+    Args:
+        client: The Flask test client.
+        name: The name to assign to the new queue.
+
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = register_queue(client, name=name)
+    assert response.status_code == 400
+
+
+def assert_queue_name_matches_expected_name(
+    client: FlaskClient, queue_id: int, expected_name: str
+) -> None:
+    """Assert that the name of a queue matches the expected name.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to retrieve.
+        expected_name: The expected name of the queue.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the name of the
+            queue does not match the expected name.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response.get_json()["name"] == expected_name
+
+
+def assert_queue_is_not_found(
+    client: FlaskClient,
+    queue_id: int,
+) -> None:
+    """Assert that a queue is not found.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to retrieve.
+
+    Raises:
+        AssertionError: If the response status code is not 404.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 404
+
+
+def assert_queue_count_matches_expected_count(
+    client: FlaskClient,
+    expected: int,
+) -> None:
+    """Assert that the number of queues matches the expected number.
+
+    Args:
+        client: The Flask test client.
+        expected: The expected number of queues.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the number of
+            queues does not match the expected number.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}",
+        follow_redirects=True,
+    )
+    assert len(response.get_json()["data"]) == expected
+
+
+def assert_cannot_rename_queue_with_existing_name(
+    client: FlaskClient,
+    queue_id: int,
+    existing_name: str,
+) -> None:
+    """Assert that renaming a queue with an existing name fails.
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to rename.
+        name: The name of an existing queue.
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = rename_queue(
+        client=client,
+        queue_id=queue_id,
+        new_name=existing_name,
+    )
```

Comment:
Needs a `group_id`

Reply:
This is a rename/modify, not a register. Why do we need `group_id`?